plugins/builds/index.js

```diff
@@ -401,7 +401,10 @@ async function getFinishedBuilds(event, buildFactory) {
  */
 async function updateParentBuilds({ joinParentBuilds, nextBuild, build }) {
     // Override old parentBuilds info
-    const newParentBuilds = merge({}, joinParentBuilds, nextBuild.parentBuilds, (objVal, srcVal) => objVal || srcVal);
+    const newParentBuilds = merge({}, joinParentBuilds, nextBuild.parentBuilds, (objVal, srcVal) =>
+        // passthrough objects, else mergeWith mutates source
+        srcVal && typeof srcVal === 'object' ? undefined : objVal || srcVal
```

Comment:
```
(srcVal && typeof srcVal === 'object' ? undefined : objVal) || srcVal
```

If `typeof srcVal === 'object'` then we use `undefined`, otherwise use `objVal`? Not sure the intent here, it just sounds a little off.

Reply:
https://lodash.com/docs/4.17.15#mergeWith

` If customizer returns undefined, merging is handled by the method instead` if value is object default mergeWith functionality. Else `mergeWith` will mutate source object